### PR TITLE
fix: resolve data file field handler

### PR DESCRIPTION
**Pull request by abhishek305**
[workerB](https://workerb.linearb.io/v2/badge/collaboration-page?magicLinkId=8CTvltx)

Changes: 
- Fix issue: https://contentstack.atlassian.net/browse/MKT-7918
- package version update: v2.1.1
- resolved handler update

### CODE REVIEW

File: src/fieldModifierLocation/field.ts [hunk 1]
```diff
@@ -24,7 +24,7 @@ function separateResolvedData(
             resolvedData = value;
             unResolvedData =
                 field.schema.multiple === true
-                    ? value.map((file: any) => file.uid)
+                    ? value.map(({ file }: any) => file.uid)
```
Comment (MaitriGurey1) [inline src/fieldModifierLocation/field.ts:27]: @abhishek305  would be better if we can have a proper type instead of any

**Review by MaitriGurey1 — APPROVED** [review]

File: src/fieldModifierLocation/field.ts (hunk 1 above)
Comment (abhishek305) [reply src/fieldModifierLocation/field.ts:27]: Considering the use here we can add a generic object for now @MaitriGurey1

File: src/field.ts [hunk 2]
```diff
@@ -21,7 +20,7 @@ function separateResolvedData(field: Field, value: GenericObjectType) {
             resolvedData = value;
             unResolvedData =
                 field.schema.multiple === true
-                    ? value.map((file: any) => file.uid)
+                    ? value.map(({ file }: GenericObjectType) => file.uid)
```
Comment (rijil-tr) [inline src/field.ts:23]: Can we use optional chaining?
```
file?.uid
```

Comment (abhishek305) [reply src/field.ts]: Done

**Review by rijil-tr — APPROVED** [review]
LGTM!

**Pull request merged**